File: problems/0/0/9/009.c
// Copyright (C) 2013 Jorge Aparicio

#include <stdio.h>

static int P = 1000;

int main() {
  int a, b, c;

  for (c = P / 3 + 1; c < P / 2; c++) {
    for (b = (P - c) / 2 + 1; b < c; b++) {
      a = P - b - c;

      if (a*a + b*b == c*c) {
        printf("%d\n", a * b * c);

        return 0;
      }
    }
  }
}

File: problems/0/0/9/009.rs
// Copyright (C) 2014 Jorge Aparicio

use std::num::pow;

static P: uint = 1000;

fn main() {
    for c in range(P / 3 + 1, P / 2) {
        for b in range((P - c) / 2 + 1, c) {
            let a = P - b - c;

            if pow(a, 2) + pow(b, 2) == pow(c, 2) {
                println!("{}", a * b * c);

                return;
            }
        }
    }
}

